src/MapWindow/GlueMapWindowOverlays.cpp

```diff
@@ -271,10 +271,20 @@ void
 GlueMapWindow::DrawMapScale(Canvas &canvas, const PixelRect &rc,
                             const MapWindowProjection &projection) const
 {
-  RenderMapScale(canvas, projection, rc, look.overlay);
 
-  if (!projection.IsValid())
-    return;
+  PixelRect scaleRef(rc.left, rc.top, rc.right, rc.bottom);
+
+  /*
+    If buttons are drawn on bottom (Pan mode and portrait);
+    Change reference for rendering so scale is drawn above.
+  */
+  if (IsPanning() && !Layout::landscape)
+    scaleRef.bottom -= rc.GetHeight() / 6; // Scale factor as in MenuBar.cpp
```

Comment:
Don't duplicate this formula. If somebody ever changes the button layout in `MenuBar.cpp`, this will break. This is very fragile code!

Reply:
I changed that in the new commit. It does however introduce ```MenuBar.hpp``` as a new dependency and makes the ```GetButtonPosition(..)``` no longer static.